### DISCUSSION TITLE
Data table using tanstack query?

### DISCUSSION THREAD

**Discussion by gmcamposano** [shadcn-ui/ui#1882] [Q&A]
Hello,
Has anyone been able to successfully build a data table using tanstack query. I keep getting typescript errors all over the place. The problem comes from TData when i am assigning the data. Also, it seems the only way to pass the data is using a dhydrated state with a hydration boundary around the TableData component.

**Comment by sadeghbarati** [comment]
Check this example might help

https://www.material-react-table.com/docs/examples/react-query

**Reply by gmcamposano** [reply]
Thank you for sharing that. I will take a look at it.
